src/credentials.rs

```diff
@@ -269,7 +269,8 @@ impl AWSCredentialsProvider for IAMRoleCredentialsProvider {
             //println!("Calling IAM metadata");
             // for "real" use: http://169.254.169.254/latest/meta-data/iam/security-credentials/
             let mut address : String = "http://169.254.169.254/latest/meta-data/iam/security-credentials".to_string();
-            let client = Client::new();
+            let mut client = Client::new();
+            client.set_read_timeout(Some(StdDuration::new(15, 0)));
```

Comment:
Perhaps we should use from_secs instead, it's clearer:

https://doc.rust-lang.org/std/time/struct.Duration.html#method.from_secs